x11-packages/synaptic/common-rpackagelister.cc.patch

```diff
@@ -27,3 +29,19 @@
        (!_xapianDatabase )) {
        if(_config->FindB("Debug::Synaptic::Xapian",false))
  	 std::cerr << "xapain index not build yet" << std::endl;
+@@ -924,11 +922,11 @@ void RPackageLister::getSummary(int &held, int &kept, int &essential,
+ 
+ 
+ 
+-struct bla:public binary_function<RPackage *, RPackage *, bool> {
+-   bool operator() (RPackage *a, RPackage *b) {
++auto bla() {
++   return [](RPackage* a, RPackage* b) -> bool {
```

Comment:
That workaround has been sanctioned by @tstein,
with the caveat and the entire Synaptic codebase is 20+ years old and is showing that age.